Client/src/Components/DatasetUpload/DatasetView.tsx

```diff
@@ -68,7 +71,11 @@ export const DatasetView = (props: IProps) => {
   }, [])
 
   const handleSubmitForm = async (formDataset: IDatasetModel) => {
-    await callSaveDataset(formDataset)
+    const result = await callSaveDataset(formDataset)
+    if (result > 0) {
+      SnackbarUtils.success("Dataset successfully uploaded")
+    }
+    history.push('/dataset/:' + formDataset.id)
```

Comment:
This part does not redirect to the newly created dataset. formDataset.id is undefined. You should pass the result as the id like so:
const handleSubmitForm = async (formDataset: IDatasetModel) => {
    const result = await callSaveDataset(formDataset)
    if (result > 0) {
      SnackbarUtils.success("Dataset successfully uploaded")
      history.push('/dataset/' + result)
      history.go(0)
    }
  }

Reply:
done